src/examples/zkapps/voting/demo.ts

```diff
@@ -0,0 +1,326 @@
+// used to do a dry run, without tests
+// ./run ./src/examples/zkapps/voting/demo.ts
+
+import {
+  Field,
+  Mina,
+  Party,
+  PrivateKey,
+  Permissions,
+  UInt64,
+  Experimental,
+  UInt32,
+} from 'snarkyjs';
+import { VotingApp, VotingAppParams } from './factory';
+import { Member, MerkleWitness } from './member';
+import { OffchainStorage } from './off_chain_storage';
+import {
+  ParticipantPreconditions,
+  ElectionPreconditions,
+} from './preconditions';
+
+let Local = Mina.LocalBlockchain();
+Mina.setActiveInstance(Local);
+
+let feePayer = Local.testAccounts[0].privateKey;
+
+let tx;
+
+// B62qra25W4URGXxZYqYjfkXBa6SfwwrSjX2ZFJ24x12sSy8khGRcRH1
+let voterKey = PrivateKey.fromBase58(
+  'EKEgiGWBmGG77ERKU7ihArYbUTfroEr466Gs1RKUph8bgpvF5BSD'
+);
+// B62qohqUFi8iy5mA4roZDNEuHdj1bWtyriYZouybC33wb8Q6AiUc7D7
+let candidateKey = PrivateKey.fromBase58(
+  'EKELdqBuWoNa4KFibyumCJNCr1SzMFJi5mV3pCASXfNH3geh6ezG'
+);
+// B62qq2s61y9gzALPWSAFitucxq1PhLEjQLGwb65gQ7UgsVFNTtjrzRj
+let votingKey = PrivateKey.fromBase58(
+  'EKFHGpCJTuQk1xHTkQH3q3xXJCHMQLPwhy5iTJk3L2bK4FG9iVnv'
+);
+
+let params: VotingAppParams = {
+  candidatePreconditions: ParticipantPreconditions.default,
+  voterPreconditions: ParticipantPreconditions.default,
+  electionPreconditions: ElectionPreconditions.default,
+  voterKey,
+  candidateKey,
+  votingKey,
+  doProofs: false,
+};
+
+let contracts = await VotingApp(params);
+
+let voterStore = new OffchainStorage<Member>(8);
+let candidateStore = new OffchainStorage<Member>(8);
+let votesStore = new OffchainStorage<Member>(8);
+
+let initialRoot = voterStore.getRoot();
+try {
+  tx = await Mina.transaction(feePayer, () => {
```

Comment:
isn't this too many actions to put into one txn (if this is executed on-chain)? I know it will work for the local blockchain case, but I wonder if we should keep these as consistent as we can with what we could do on-chain too

Reply:
yep, we can! This is also only the "demo" script, to demonstrate a basic run through - the tests will look better